src/libAtomVM/CMakeLists.txt

```diff
@@ -87,15 +85,40 @@ set(SOURCE_FILES
     timer_wheel.c
     valueshashtable.c
 )
+
+add_library(libAtomVM ${SOURCE_FILES} ${HEADER_FILES})
+target_include_directories(libAtomVM PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
+target_compile_features(libAtomVM PUBLIC c_std_11)
+if(CMAKE_COMPILER_IS_GNUCC)
+    target_compile_options(libAtomVM PUBLIC -Wall -pedantic -Wextra -ggdb)
+endif()
+
+# AtomVM options used in libAtomVM
+if (ADVANCED_TRACING)
+    target_compile_definitions(libAtomVM PUBLIC ENABLE_ADVANCED_TRACE)
+endif()
+
+if (AVM_DISABLE_FP)
+    target_compile_definitions(libAtomVM PUBLIC AVM_NO_FP)
+else()
+    target_link_libraries(libAtomVM PUBLIC m)
+endif()
+
+if (AVM_USE_32BIT_FLOAT)
```

Comment:
Probably not already an issue and likely harmless, but it might read more cleanly if this test was only done if `AVM_DISABLE_FP` not set.

Reply:
You mean we should introduce some logic in the options and error if user select both disable FP and 32 bits floats?
I still don't understand the value of disabling FP since the compiler is likely to generate FP opcodes, but this is another story.

Reply:
Yes, adding logic for that condition sounds tedious and wasted effort of limited resources, if we want to remove the option, anyway.

Reply:
I'll add an error message for a better UX, and we'll remove it when getting rid of DISABLE_FP.